src/backend/opencl/program.hpp

```diff
@@ -32,21 +34,26 @@
 #define SHOW_BUILD_INFO(PROG)                                              \
     do {                                                                   \
         std::string info = getEnvVar("AF_OPENCL_SHOW_BUILD_INFO");         \
-        if (!info.empty() && info != "0") { SHOW_DEBUG_BUILD_INFO(prog); } \
+        if (!info.empty() && info != "0") { SHOW_DEBUG_BUILD_INFO(PROG); } \
     } while (0)
 
 #else
 #define SHOW_BUILD_INFO(PROG) SHOW_DEBUG_BUILD_INFO(PROG)
 #endif
 
-namespace cl {
-class Program;
-}
-
 namespace opencl {
-void buildProgram(cl::Program &prog, const char *ker_str, const int ker_len,
+
+//TODO(pradeep) remove this version after porting to new cache interface
+[[deprecated("use cl::Program buildProgram(vector<string>&, vector<string>&)")]]
```

Comment:
Can we add an ignore to the warnings? Or make the [deprecated] optional based on a build flag. Although this is the correct way and would be useful to find all of places where we need to update the api, it'd be nice to avoid a ton of build warnings for now

Reply:
Adding `ignore` would defeat the purpose of the attribute completely. We can keep this change off from next release (3.7) so that we have enough time to port everything to new cache API. That way, 3.7.* builds won't have any warnings whoever tries to build it. warnings on master fine I think since it is development branch and since these are not some invalid logic related warnings.

Reply:
Having too many warnings will hide warnings that are important and may be new. We can deprecate these but that warning should be enabled explicitly in case you want to search for their usage. Although grepping would work better in that case.

Reply:
I have an idea to do this. I don't think grep is great, it involves additional post-processing again.